### PR TITLE
chore(sdk): Adds Rust example for message compression using user_headers

### LINKED ISSUE #30: Client-side message compression feature
This feature would add optional client-side compression to reduce network bandwidth between clients and servers.

Proposed Implementation

1. Use custom message headers to indicate compression (see field `user_headers` in `IggyMessage`):
```rust
const COMPRESSION_HEADER_KEY: &str = "iggy-compression";  // Values: "gzip", "zstd", "lz4", "snappy"
```

2. Add compression config to producer:
```rust
  pub struct ClientCompressionConfig {
      pub algorithm: ClientCompressionAlgorithm,  // Gzip, Zstd, Lz4, Snappy
      pub min_size: u32,  // Only compress if payload > min_size (default: 1KB)
  }
```
3. Producer compresses paylaod before sending, adds custom user header to `IggyMessage` (K:V)
4. Consumer detects header and decompresses transparently

Tasks

- Add compression deps to Cargo.toml (flate2, zstd, lz4, snap) with feature flags
- Implement compress/decompress in producer/consumer message flow
- Add tests for different algorithms and edge cases
- Benchmark bandwidth vs CPU trade-offs using `iggy-bench` (also implement handling of compression for it)

### PR DESCRIPTION
**Pull request by haubur**
# Rust example: Per message compression using user_headers
Closes #30

- Consolidates exisiting example `message-headers` into `message-headers/message-type`
- Adds new example to `message-headers/message-compression`
- Adds a producing example `message-headers/message-compression/producer/main.rs`
- Adds a corresponding consuming example `message-headers/message-compression/consumer/main.rs` 
- Adds `codec.rs` to `examples/rust/src/shared` that implements shared utilities to compress and decompress messages along with methods to transparently resolve user_headers into a codec.
- Adds README.md with some more details, to keep code examples readable.

### CODE REVIEW

File: examples/rust/src/message-headers/message-compression/README.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+Details on how to run the examples for the Rust Iggy SDK can be found in the parent folder [README.md](https://github.com/apache/iggy/tree/master/examples/rust#readme).
+
+Run the following commands
+
+1. Set root credentials in your environment
+
+  ```bash
+  export IGGY_ROOT_USERNAME=iggy
+  export IGGY_ROOT_PASSWORD=iggy
+  ```
```
Comment (hubcio) [inline examples/rust/src/message-headers/message-compression/README.md]: you can skip these if you start server with `--with-default-root-credentials` (which you do)

File: examples/rust/src/message-headers/message-compression/README.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+
+1. Start the server
+
+  ```bash
+  cargo run --bin iggy-server -- --with-default-root-credentials
+  ```
+
+  **NOTE**: In case the server was running before, make sure to run `rm -rf local_data/` to delete server state data from prior runs.
+
+1. Run the producer to write compressed messages to the server
```
Comment (hubcio) [inline examples/rust/src/message-headers/message-compression/README.md]: all points are starting with `1.`

File: examples/rust/src/message-headers/message-compression/producer/main.rs [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    // Login using default credentials.
+    client
+        .login_user(DEFAULT_ROOT_USERNAME, DEFAULT_ROOT_PASSWORD)
+        .await?;
+
+    // Create a Stream.
+    client
+        .create_stream(STREAM_NAME)
+        .await
+        .expect("Stream was NOT created! Start a fresh server to run this example.");
```
Comment (hubcio) [inline examples/rust/src/message-headers/message-compression/producer/main.rs]: you can mention `--fresh` argument to `iggy-server` binary

File: examples/rust/src/shared/codec.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+                let mut compressed_data = Vec::new();
+                let mut encoder = FrameEncoder::new(&mut compressed_data);
+                encoder
+                    .write_all(data)
+                    .expect("Cannot write into buffer using Lz4 compression.");
+                encoder.finish().expect("Cannot finish Lz4 compression.");
+                Ok(compressed_data)
+            }
+        }
+    }
```
Comment (hubcio) [inline examples/rust/src/shared/codec.rs:91]: this functions returns Result but in fact it never does because `expect()` is used

either simplify and return `Vec<u8>` (which it totally fine for example or propagate errors above

File: examples/rust/src/message-headers/message-compression/README.md [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+  ```bash
+  cargo run --example message-headers-compression-consumer
+  ```
+
+## The Codec
+
+The **co**mpression and **dec**compression utilities are implemented in `examples/rust/src/shared/codec.rs` and used when sending messages to the server and reading them from the server.
+
+First, define a stream and a topic name.
+The producer will first initiate the stream and the topic on that stream and then write the example messages to that topic wihtin that stream.
```
Comment (hubcio) [inline examples/rust/src/message-headers/message-compression/README.md]: typo: `within`

File: examples/rust/src/message-headers/message-compression/README.md [hunk 6]
```diff
@@ -0,0 +1,269 @@
...
+                encoder
+                    .write_all(data)
+                    .expect("Cannot write into buffer using Lz4 compression.");
+                encoder.finish().expect("Cannot finish Lz4 compression.");
+                Ok(compressed_data)
+            }
+        }
+    }
+
+    pub fn decompress(&self, data: &[u8]) -> Result<Vec<u8>, IggyError> {
```
Comment (hubcio) [inline examples/rust/src/message-headers/message-compression/README.md]: this is incorrect, in code it actually returning Vec<u8>
